### ISSUE TITLE
On disconnect unsubscribe is not triggered

### ISSUE THREAD

**Issue by Aegoon** [emqx/emqx-rule-engine#125]
**Environment**

    OS: Ubuntu 16.04
    EMQ: 3.2.3

**Description**

The intention is capture when specific device is no longer subscribed to a topic.
But if device is loosing internet connection for example only event.disconnected is triggered but not event.unsubscribe.

The reason for using unsubscribe is because we don't want to trigger engine rule for all connected devices and sql 'LIKE' operator doesn't work for filter by field (clientId for example).
So we tried to use unsubscribe filtering by topic and found this issue (if this is actually an issue).

You may not consider this as a bug because it may be functional but it doesn't seems like correct behavior.

btw Rule Engine is great feature, make life much easier.

**Comment by terry-xiaoyu** [comment]
@Aegoon May I ask your requirement in more details, for example, what kind of rule SQL do you like?

We provided some string processing functions:

```erlang
-export([ lower/1
        , ltrim/1
        , reverse/1
        , rtrim/1
        , strlen/1
        , substr/2
        , substr/3
        , trim/1
        , upper/1
        , split/2
        , split/3
        ]).
```

Like this:
```sql
select split(client_id, '-') as fullname

from "message.publish"

where nth(1, fullname) = 'steve'
```

And if given a client_id = 'steve-jobs'

The SQL would outputs:
```json
{
  "fullname": [
    "steve",
    "jobs"
  ]
}
```

For the unsubscribe event, it won't be triggered if a client is disconnected, that's the correct behavior I think.

**Comment by Aegoon** [comment]
> For the unsubscribe event, it won't be triggered if a client is disconnected, that's the correct behavior I think.

I suspected it.

split will solved perfectly the use case, thank you.
